gateway/http.d/init.conf

```diff
@@ -12,6 +12,13 @@ init_by_lua_block {
     local context = env:context()
 
     local resty_env = require('resty.env')
+    -- this file both liquid template and pure nginx file
+    -- {% raw %} and {{ ENV }} is both valid Lua code and Liquid variable interpolation {% endraw %}
+    -- so we can push ENV from outside when this file is included and inject extra environment
```

Comment:
@davidor I definitely have to improve this and properly document it, because this is serious mindfuck.

I was thinking about something like:
```lua
-- WARNING, WARNING, WARNING: this is insane hack and should not be touched
-- This file is not templated by Liquid, because it is used by not blackbox integration tests (Search for TEST_NGINX_HTTP_CONFIG).
-- So it can't be templated by liquid, to lift env variables and persist them in the configuration.
-- This is a workaround so we can store environment at the time of building the config into the config itself.
--  {% raw %}  {{ ENV }}  {% endraw %} is both valid Liquid template and Lua code.
-- In Lua it is a table with another empty table inside. In Liquid it prints variable ENV.
-- This ENV variable is defined in the main nginx.conf.liquid and injected when including this partial.
-- The content of the ENV variable is a Lua table, so when rendered it actually can run ipairs on it.
```